### PR TITLE
Np 47011 brage merging search api and dynamo db mismatch handling

### PR DESCRIPTION
**Pull request by anetteOlli**

### CODE REVIEW

File: publication-commons/src/main/java/no/unit/nva/publication/model/SearchResourceApiResponse.java [hunk 1]
```diff
@@ -14,6 +14,8 @@ public String toString() {
         return toJsonString();
     }
 
+
+    @JacocoGenerated
```
Comment (anetteOlli) [inline publication-commons/src/main/java/no/unit/nva/publication/model/SearchResourceApiResponse.java:18]: Denn brukes fortsatt i bragePatcheren. Trengs denne fortsatt egentlig?

File: brage-import/src/test/java/no/sikt/nva/brage/migration/lambda/BrageEntryEventConsumerTest.java [hunk 2]
```diff
@@ -1895,6 +1896,40 @@ void shouldMergeConferenceReportWithEvent() throws IOException {
...
+        var instanceType = existingPublication.getEntityDescription().getReference().getPublicationInstance().getInstanceType();
+        var contributor = existingPublication.getEntityDescription().getContributors().getFirst();
+        var brageContributor = new Contributor(new Identity(contributor.getIdentity().getName(), null),
+                                               "ARTIST", null, List.of());
+        var generator = new NvaBrageMigrationDataGenerator.Builder()
+                            .withMainTitle("Dynamic Response of Floating Wind Turbines")
+                            .withContributor(brageContributor)
+                            .withPublicationDate(new PublicationDate("2023",
+                                                                     new PublicationDateNva.Builder().withYear("2023").build()))
+                            .withType(new Type(List.of(), instanceType)).build();
```
Comment (truhacevkir) [inline brage-import/src/test/java/no/sikt/nva/brage/migration/lambda/BrageEntryEventConsumerTest.java:1918]: We have a method for persisting duplicate

`var generator = generateBrageRecordAndPersistDuplicate(new Lecture(), TYPE_CONFERENCE_REPORT);`

**Review by truhacevkir — APPROVED** [review]

File: brage-import/src/test/java/no/sikt/nva/brage/migration/lambda/BrageEntryEventConsumerTest.java (hunk 2 above)
Comment (anetteOlli) [reply brage-import/src/test/java/no/sikt/nva/brage/migration/lambda/BrageEntryEventConsumerTest.java:1918]: nice will change.

File: publication-commons/src/main/java/no/unit/nva/publication/model/SearchResourceApiResponse.java (hunk 1 above)
Comment (truhacevkir) [reply publication-commons/src/main/java/no/unit/nva/publication/model/SearchResourceApiResponse.java:18]: Kan sikkert fjernes ja

**Pull request merged**